include/picongpu/fields/MaxwellSolver/Yee/Yee.kernel

```diff
@@ -111,127 +122,75 @@ namespace picongpu
                  */
-                template<uint32_t T_numWorkers, typename T_BlockDescription>
-                struct KernelUpdateE
+                template<uint32_t T_numWorkers>
+                struct KernelUpdateField
                 {
                     /** Update the electric field using the given functor
                      *
                      * @tparam T_Acc alpaka accelerator type
                      * @tparam T_Mapping mapper functor type
-                     * @tparam T_UpdateFunctor update functor type
-                     * @tparam T_BBox pmacc::DataBox, magnetic field box type
-                     * @tparam T_EBox pmacc::DataBox, electric field box type
+                     * @tparam T_StencilFunctor stencil functor type to update a cell
+                     * @tparam T_SrcBox pmacc::DataBox, source field box type
+                     * @tparam T_DestBox pmacc::DataBox, electric field box type
                      *
                      * @param acc alpaka accelerator
                      * @param mapper functor to map a block to a supercell
-                     * @param updateFunctor update functor
-                     * @param fieldB magnetic field iterator
-                     * @param fieldE electric field iterator
+                     * @param stencilFunctor stencil functor
+                     * @param srcField source field iterator (is not allowed to be an alias of destField data)
+                     * @param destField destination field iterator
                      */
                     template<
                         typename T_Acc,
                         typename T_Mapping,
-                        typename T_UpdateFunctor,
-                        typename T_BBox,
-                        typename T_EBox>
+                        typename T_StencilFunctor,
+                        typename T_SrcBox,
+                        typename T_DestBox>
                     DINLINE void operator()(
                         T_Acc const& acc,
                         T_Mapping const mapper,
-                        T_UpdateFunctor updateFunctor,
-                        T_BBox const fieldB,
-                        T_EBox fieldE) const
+                        T_StencilFunctor stencilFunctor,
+                        T_SrcBox const srcField,
+                        T_DestBox destField) const
                     {
-                        /* Each block processes grid values in a supercell,
+                        /* Each block processes all cells of a supercell,
                          * the index includes guards, same as all indices in this kernel
                          */
-                        auto const blockBeginIdx = mapper.getSuperCellIndex(DataSpace<simDim>(cupla::blockIdx(acc)))
+                        auto const superCellIdx = mapper.getSuperCellIndex(DataSpace<simDim>(cupla::blockIdx(acc)))
```

Comment:
But it is the index of the supercell, therefore I renamed it. Block index is wrong because it is `block index  + guard supercells`. This means the alpaka block index is already mapped into the super cell index domain.

Reply:
It is not. Index of the supercell is the first multiplier. It is then multiplied by the supercell size

Reply:
I think you introduced this bug in the PR, see how it was defined before